File: androidfun/net/code/com/net/model/chick/course/CourseInfo.java
package com.net.model.chick.course;

import java.util.List;

/**
 * 佛祖保佑         永无BUG
 *
 * @author Created by joker on 2020/7/8
 */
public class CourseInfo {

    /**
     * 课程主键id
     */
    public int courseId;

    /**
     * 主播id
     */
    public int userId;

    /**
     * 课程名称
     */
    public String courseName;

    /**
     * 课程简介
     */
    public String courseIntroduction;

    /**
     * 课程亮点
     */
    public String courseHighlights;

    /**
     * 售价(钻)
     */
    public int cost;

    /**
     * 划线原价
     */
    public int crossedPrice;

    /**
     * 是否可以用
     */
    public boolean inUse;

    /**
     * 总讲数
     */
    public int audioCount;

    /**
     * 形象照
     */
    public String personalImage;
    public String backgroundColor;

    /**
     * 章节列表
     */
    public List<CourseChapter> courseChapterList;

    /**
     * 讲列表(没有章节的情况)
     */
    public List<AudioInfo> courseChapterAudioList;
}

File: androidlib/im_v4/function/model/code/com/joker/im/custom/chick/CourseInfo.kt
package com.joker.im.custom.chick

import java.io.Serializable

/**
 *
 * @author Created by lenna on 2020/7/6
 */
class CourseInfo : Serializable {
    /**课程id*/
    var courseId = 0

    /**主播id*/
    var userId = 0

    /**课程名称*/
    var courseName = ""

    /**课程售价（单位砖）*/
    var cost = 0

    /**
     * 售价富文本
     */
    var priceText: List<SplitText> = arrayListOf()

    /**原价（单位砖）*/
    var crossedPrice = 0

    /**总讲数*/
    var audioCount = 0

    /**是否可用*/
    var inUse: Boolean = false

    /**是否已购买*/
    var hasBuy: Boolean = false

    /**免费领取*/
    var hasReceive: Boolean = false

    /**
     * 是否是自己
     */
    var isSelf: Boolean = false

    var cardType = CardType.TYPE_NORMAL

    var backgroundColor = ""

    // 是否有付费的音频
    var hasFreeAudition: Boolean = false
    var isGroupMember: Boolean = false

    // 课程当前工作室roomId
    var roomId: String? = null

    // 课程当前工作室groupId
    var groupId: Int = 0

    object CardType {
        const val TYPE_NORMAL = "normal"
        const val TYPE_CANCEL = "cancel"
        const val TYPE_RECOMMEND = "recommend"
        const val TYPE_RECHARGE = "recharge"
        const val TYPE_ASK = "ask"
    }


}